shared/android/app/src/main/java/io/keybase/ossifrage/KeybasePushNotificationListenerService.java

```diff
@@ -288,3 +238,70 @@ public void add(MessagingStyle.Message m) {
         return buffer;
     }
 }
+
+class NotificationData {
+    final boolean displayPlaintext;
+    final int membersType;
+    final String convID;
+    final int messageId;
+    final String pushId;
+    final int badgeCount;
+    final long unixTime;
+    final String soundName;
+    final String serverMessageBody;
+
+    // Derived from go/gregord/chatpush/push.go
+    @RequiresApi(api = Build.VERSION_CODES.HONEYCOMB_MR1)
+    NotificationData(String type, Bundle bundle) {
+        displayPlaintext = "true".equals(bundle.getString("n"));
+        membersType = Integer.parseInt(bundle.getString("t"));
+        badgeCount = Integer.parseInt(bundle.getString("b", "0"));
+        soundName = bundle.getString("s", "");
+        serverMessageBody = bundle.getString("message", "");
+
+        if (type.equals("chat.newmessage")) {
+            messageId = Integer.parseInt(bundle.getString("msgID", "0"));
+            convID = bundle.getString("convID");
+            // TODO consolidate this with below when new server code goes in
+            unixTime = ((new Date()).getTime() / 1000);
+        } else if (type.equals("chat.newmessageSilent_2"))  {
+            messageId = Integer.parseInt(bundle.getString("d", ""));
+            convID = bundle.getString("c");
+            unixTime = Long.parseLong(bundle.getString("x", "0"));
+        } else {
+            throw new Error("Tried to parse notification of unhandled type: " + type );
+        }
+
+        String pushIdTmp = "";
```

Comment:
this should happen in the silent parsing branch, will always fail for newmessage